docs/troubleshooting/monitoring.md

```diff
@@ -316,3 +316,284 @@ You may encounter this when you install the Harvester v1.3.0 or higher version c
+    app.kubernetes.io/managed-by: Helm
+  name: rancher-monitoring-crd-manager
+  namespace: cattle-monitoring-system
+  ...
+```
+
+### Root Cause
+
+The objects that are marked as `missing` do not have the related labels and annotations required by the `ManagedChart` object.
+
+Example:
+
+```
+One of the manually recreated object:
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  annotations:
+    meta.helm.sh/release-name: rancher-monitoring-crd
+    meta.helm.sh/release-namespace: cattle-monitoring-system
+    objectset.rio.cattle.io/id: default-mcc-rancher-monitoring-crd-cattle-fleet-local-system   # This required item is not in the above object.
+  creationTimestamp: "2024-04-03T10:23:55Z"
+  labels:
+    app: rancher-monitoring-crd-manager
+    app.kubernetes.io/managed-by: Helm
+    objectset.rio.cattle.io/hash: 2da503261617e9ea2da822d2da7cdcfccad847a9    # This required item is not in the above object.
+  name: rancher-monitoring-crd-manager
+...
+rules:
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - create
+  - get
+  - patch
+  - delete
+  - update
+```
+
+### Workaround
+
+To fix the issue, you must recreate the `rancher-monitoring-crd` Managedchart object.
+
+1. Verify the chart version of `rancher-monitoring-crd` using the `harvester-cluster-repo` service.
+
+    Get the IP of the `harvester-cluster-repo` service.
+
+    ```
+    $ kubectl get service -A | grep repo
+    cattle-system                     harvester-cluster-repo                        ClusterIP      10.53.146.66    <none>            80/TCP                         8d
+
+    The IP address is 10.53.146.66.
+    ```
+
+    Get the chart version from the `index.yaml` file in the `harvester-cluster-repo`.
+
+    ```
+    ssh to Harvester VIP, then $ curl http://10.53.146.66/charts/index.yaml | grep "rancher-monitoring"
+    ...
+    name: rancher-monitoring
+        - rancher-monitoring-102.0.0+up40.1.2.tgz
+      rancher-monitoring-crd:
+          catalog.cattle.io/release-name: rancher-monitoring-crd
+        description: Installs the CRDs for rancher-monitoring.
+        name: rancher-monitoring-crd
+        - rancher-monitoring-crd-102.0.0+up40.1.2.tgz
+
+    The chart version is 102.0.0+up40.1.2.
+    ```
+
+    The chart version must be identical to the value of the `version` field in the `rancher-monitoring-crd` ManagedChart object.
+
+    :::note
+
+    If those two versions are different, you need to stop and ask help from Harvester support or [community](https://github.com/harvester/harvester?tab=readme-ov-file#community).
+
+    :::
+
+1. Disable the `rancher-monitoring` [addon](../advanced/addons.md). Wait until the state changes to `Disabled`.
+
+1. Delete the `rancher-monitoring-crd` ManagedChart.
+
+   ```
+   $ kubectl delete managedchart -n fleet-local rancher-monitoring-crd
+   ```
+
+1. Manually delete all objects marked as `missing`..
+
```

Comment:
Oh I meant users might have been enabled the `rancher-monitoring` add-on and created some custom resources like alertmanagerconfigs. Will they stay intact after the `rancher-monitoring-crd` ManagedChart was removed? If the answer is no, then it might be necessary to remind users to backup those custom resources first.

Reply:
A good point, I will double check and validate, thanks.

Reply:
Based on this important comment, we rebased the workaroud to patch the missing annotation and label manually, without deleting the managedchart and recreating it.